src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -302,6 +317,12 @@ private static AnalyzerConfigDocument CreateAnalyzerConfigDocument(DocumentId do
             return new AnalyzerConfigDocument(project, state);
         }
 
+        private static readonly Func<DocumentId, (SourceGeneratedDocumentState, Project), SourceGeneratedDocument> s_createSourceGeneratedDocumentFunction = CreateSourceGeneratedDocument;
+        private static SourceGeneratedDocument CreateSourceGeneratedDocument(DocumentId documentId, (SourceGeneratedDocumentState, Project) stateAndProject)
+        {
+            return new SourceGeneratedDocument(stateAndProject.Item2, stateAndProject.Item1);
+        }
```

Comment:
can this method (and s_createSourceGeneratedDocumentFunction ) just be removed.  Just make a static lambda at the use site?  I think this pattern was fine *prio* to us having static lambdas.  but now that we do, it just feels unnecessary and verbose.

Reply:
We are using it multiple places, so there's no single place to inline the code to -- also even if it's a static lambda isn't that still allocating on each use? I believe the pattern here is to avoid the allocations as well.

Reply:
>  also even if it's a static lambda isn't that still allocating on each use?

Nope!  the static lambda becomes a single allocation for the lifetime of the program.  Very very handy for when you want this sort of pattern, but do not want GC costs.

Reply:
(of course, if you use it in multiple places, there's nothing you can do).